colcon_sanitizer_reports/_sanitizer_section_part_stack_trace.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from typing import Tuple
+
+
+# Key comes from a line of ros2 code and matches the following pattern.
+_FIND_KEY_REGEX = re.compile(r'^\s*#\d+ (0x[\da-f]+ in|)\s*(?P<key>.*/ros2.*)\s*$')
+
+# Parts of a key that are changeable between otherwise identical stack trace can be found and masked
+# with the following pattern.
+_FIND_KEY_SUB_REGEX = re.compile(r'0x[\da-f]+')
+
+
+class SanitizerSectionPartStackTrace:
```

Comment:
It seems like this class exists to produce a key string. Is there a strong reason to prefer a class over a function?

Reply:
Yes. I was trying to set myself up for an easy follow-up PR where I need to store the lines of the stack trace as well.